sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64 } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+export interface PasskeyCreateOptions {
+	timeout: number; // The timeout for the passkey creation dialog.
+	displayName: string; // The display name of the user, this is shown at creation dialog
+	rpName: string; // The name of the relying party, this is shown to user in the passkey creation dialog.
+}
+
+export interface PasskeySignOptions {
+	timeout: number; // The timeout for the passkey authentication dialog.
+}
+
+function validateCreateOptions(options: PasskeyCreateOptions) {
+	if (options.timeout <= 0) throw new Error('Timeout must be positive');
+	if (!options.displayName?.trim()) throw new Error('Display name cannot be empty');
+	if (!options.rpName?.trim()) throw new Error('RP name cannot be empty');
+	if (options.timeout > 300000) throw new Error('Timeout cannot exceed 5 minutes');
+}
+
+export interface PasskeyProvider {
+	create(options: PasskeyCreateOptions): Promise<RegistrationCredential>;
+	get(challenge: Uint8Array, timeout: number): Promise<AuthenticationCredential>;
+}
+
+// Default browser implementation
+export class BrowserPasskeyProvider implements PasskeyProvider {
+	async create(options: PasskeyCreateOptions): Promise<RegistrationCredential> {
+		return (await navigator.credentials.create({
+			publicKey: {
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
```

Comment:
The very limited docs I've read on the passKey apis indicate the challenge is supposed to be random bytes.  What is this string based on?

Reply:
this field does not matter since we just need to retrieve the public key field. (its normally used when verifying a signature when authentication, but not needed here). we can change to a different default value or random string if thats more desirable, not any strong opinion.